testbed/tests/assertions.py

```diff
@@ -1,21 +1,44 @@
 from pytest import approx
 
 from toga.colors import TRANSPARENT
-
-
-# This could be generalized in future to accept syntax like:
-#   * assert_set_get(obj, name, pytest.approx(value)) - for floating point values
-#   * assert_set_get(obj, name, set_value, get_value) - where the two values are different
-def assert_set_get(obj, name, value):
-    """Calls a setter, then asserts that the same value is returned by the getter."""
-    setattr(obj, name, value)
-    assert getattr(obj, name) == value
+from toga.fonts import (
+    CURSIVE,
+    FANTASY,
+    MONOSPACE,
+    SANS_SERIF,
+    SERIF,
+    SYSTEM,
+)
 
 
 def assert_color(actual, expected):
-    if expected in [None, TRANSPARENT]:
+    if expected in {None, TRANSPARENT}:
         assert expected == actual
     else:
-        for component in ["r", "g", "b"]:
-            assert getattr(actual, component) == getattr(expected, component)
-        assert actual.a == approx(expected.a, abs=(1 / 255))
+        if actual in {None, TRANSPARENT}:
+            assert expected == actual
+        else:
+            assert (actual.r, actual.g, actual.b, actual.a) == (
+                expected.r,
+                expected.g,
+                expected.b,
+                approx(expected.a, abs=(1 / 255)),
+            )
+
+
+GENERIC_FONTS = {
+    CURSIVE: ["Apple Chancery"],
+    FANTASY: ["Papyrus"],
+    MONOSPACE: ["Courier New"],
+    SANS_SERIF: ["Helvetica"],
+    SERIF: ["Times", "Times New Roman"],
+    SYSTEM: [
+        SANS_SERIF,
+        ".AppleSystemUIFont",
+        "sans-serif-medium",  # Android buttons
+    ],
+}
```

Comment:
What's the reasoning behind having this table of "generic" fonts in the testbed assertions, rather than being something that is normalised out by the test backend? Previously, `toga_font` on each backend would normalise ".AppleSystemUIFont" into SYSTEM, but do so in the Cocoa backend. 

Having this table here seems like an odd inversion - we've got platform-specific details in the generic test definitions.

Reply:
The problem I was trying to solve was the many-to-one mapping between Toga font names and actual fonts. For example, SANS_SERIF and SYSTEM will often give you the same visible result, as will explicitly specifying the font name. But whether these 3 options can be distinguished by the probe varies from platform to platform.

But you're right, it would be better for platform-specific differences to be handled by the test backend, so I'll do that.